File: GremsReliquary/src/main/java/gremsReliquary/actions/BrokenRecordAction.java
package gremsReliquary.actions;

import com.megacrit.cardcrawl.actions.AbstractGameAction;
import com.megacrit.cardcrawl.actions.utility.UseCardAction;
import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.cards.CardQueueItem;
import com.megacrit.cardcrawl.core.Settings;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.monsters.AbstractMonster;

public class BrokenRecordAction extends AbstractGameAction {
    AbstractCard card;
    UseCardAction useCardAction;
    
    public BrokenRecordAction(AbstractCard card, UseCardAction useCardAction) {
        this.card = card;
        this.useCardAction = useCardAction;
        
        duration = Settings.ACTION_DUR_FAST;
    }
    
    @Override
    public void update() {
        if (duration == Settings.ACTION_DUR_FAST) {
            if (!card.purgeOnUse) {
                AbstractMonster m = null;
                if (useCardAction.target != null) {
                    m = (AbstractMonster) useCardAction.target;
                }
                
                AbstractCard tmp = card.makeSameInstanceOf();
                AbstractDungeon.player.limbo.addToBottom(tmp);
                tmp.current_x = card.current_x;
                tmp.current_y = card.current_y;
                tmp.target_x = (float) Settings.WIDTH / 2.0F - 300.0F * Settings.scale;
                tmp.target_y = (float) Settings.HEIGHT / 2.0F;
                tmp.freeToPlayOnce = true;
                if (m != null) {
                    tmp.calculateCardDamage(m);
                }
                
                tmp.purgeOnUse = true;
                AbstractDungeon.actionManager.cardQueue.add(new CardQueueItem(tmp, m, card.energyOnUse));
            }
            tickDuration();
        }
        tickDuration();
    }
}

File: GremsReliquary/src/main/kotlin/kotlinReliquary/actions/BrokenRecordAction.kt
package kotlinReliquary.actions

import com.megacrit.cardcrawl.actions.AbstractGameAction
import com.megacrit.cardcrawl.actions.utility.UseCardAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.cards.CardQueueItem
import com.megacrit.cardcrawl.core.Settings
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.monsters.AbstractMonster

class BrokenRecordAction(var card: AbstractCard, var useCardAction: UseCardAction) : AbstractGameAction() {

    init {
        duration = Settings.ACTION_DUR_FAST
    }

    override fun update() {
        if (duration == Settings.ACTION_DUR_FAST) {
            if (!card.purgeOnUse) {
                var m: AbstractMonster? = null
                if (useCardAction.target != null) {
                    m = useCardAction.target as AbstractMonster
                }

                val tmp = card.makeSameInstanceOf()
                AbstractDungeon.player.limbo.addToBottom(tmp)
                tmp.current_x = card.current_x
                tmp.current_y = card.current_y
                tmp.target_x = Settings.WIDTH.toFloat() / 2.0f - 300.0f * Settings.scale
                tmp.target_y = Settings.HEIGHT.toFloat() / 2.0f
                tmp.freeToPlayOnce = true
                if (m != null) {
                    tmp.calculateCardDamage(m)
                }

                tmp.purgeOnUse = true
                AbstractDungeon.actionManager.cardQueue.add(CardQueueItem(tmp, m, card.energyOnUse))
            }
            tickDuration()
        }
        tickDuration()
    }
}
